kits/kahuna-kit/src/main/java/com/mparticle/kits/KahunaKit.java

```diff
@@ -248,25 +253,43 @@ public void setUserIdentity(String id, MParticle.IdentityType identityType) {
         String kahunaKey;
         switch (identityType) {
             case Facebook:
-                kahunaKey = KahunaUserCredentialKeys.FACEBOOK_KEY;
+                kahunaKey = KahunaUserCredentials.FACEBOOK_KEY;
                 break;
             case Email:
-                kahunaKey = KahunaUserCredentialKeys.EMAIL_KEY;
-                break;
-            case Other:
-                kahunaKey = "user_id";
+                kahunaKey = KahunaUserCredentials.EMAIL_KEY;
                 break;
             case CustomerId:
-                kahunaKey = KahunaUserCredentialKeys.USERNAME_KEY;
+                kahunaKey = KahunaUserCredentials.USERNAME_KEY;
                 break;
             case Twitter:
-                kahunaKey = KahunaUserCredentialKeys.TWITTER_KEY;
+                kahunaKey = KahunaUserCredentials.TWITTER_KEY;
+                break;
+            case Google:
+                kahunaKey = KahunaUserCredentials.GOOGLE_PLUS_ID;
+                break;
+            case FacebookCustomAudienceId:
+                kahunaKey = "fb_app_user_id";
+                break;
+            case Microsoft:
+                kahunaKey = "msft_id";
+                break;
+            case Yahoo:
+                kahunaKey = "yahoo_id";
+                break;
+            case Other:
+                kahunaKey = "mp_other_id";
                 break;
```

Comment:
Right now they will be converted to attributes, but if we land a customer from mParticle who needs these tracked as User Credentials, we can consider adding them to our whitelist server side without going through another KahunaKit/mParticle SDK update.